src/uhi/typing/serialization.py

```diff
@@ -17,6 +17,8 @@
     "WeightedStorage",
 ]
 
+SupportedMetadata = Union[float, str, bool]
```

Comment:
```suggestion
type SupportedMetadata = float | str | bool
```

This is the Python 3.12+ syntax, by the way. :)

(PS: interesting, this is different from a Union, it's less usable at runtime, for example)